data_store/validation/pathfinders/schema_validation/validate.py

```diff
@@ -61,17 +61,19 @@ class TableValidator:
         "check": r"dtype\(.*",  # catches dtype failures
     }
 
-    def __init__(self, schema_config: dict[str, Any]) -> None:
+    def __init__(self, validate_config: ValidateConfig) -> None:
         """Initialises a TableValidator.
 
-        :param schema_config: A dictionary containing schema configuration.
+        :param validate_config: The configuration for the schema to validate against.
             The schema configuration should include information about columns, data types,
             and any additional validation rules.
-        :raises ValueError: If the schema has coercion enabled.
         """
-        if schema_config.get("coerce") or any(column.coerce for column in schema_config.get("columns", {}).values()):
```

Comment:
Do we still need the latter conditional checking for `coerce` on the columns? It's not available on the schema_config itself now but is it still avaialble on the columns?